pkg/controller/petset/pet_set_utils.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package petset
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/client/cache"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller"
+
+	"github.com/golang/glog"
+)
+
+// overlappingPetSets sorts a list of PetSets by creation timestamp, using their names as a tie breaker.
+type overlappingPetSets []apps.PetSet
+
+func (o overlappingPetSets) Len() int      { return len(o) }
+func (o overlappingPetSets) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
+
+func (o overlappingPetSets) Less(i, j int) bool {
+	if o[i].CreationTimestamp.Equal(o[j].CreationTimestamp) {
+		return o[i].Name < o[j].Name
+	}
+	return o[i].CreationTimestamp.Before(o[j].CreationTimestamp)
+}
+
+// updatePetCount attempts to update the Status.Replicas of the given PetSet, with a single GET/PUT retry.
+func updatePetCount(kubeClient *client.Client, ps apps.PetSet, numPets int) (updateErr error) {
+	if ps.Status.Replicas == numPets || kubeClient == nil {
+		return nil
+	}
+	psClient := kubeClient.Apps().PetSet(ps.Namespace)
```

Comment:
Yeah, fail. Thankfully I got the url endpoint right or it would've been a pain to fix.